src/test/kotlin/org/rust/ide/actions/RsCreateCrateActionTest.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.testFramework.MapDataContext
+import com.intellij.testFramework.TestActionEvent
+import junit.framework.Assert
+import org.rust.cargo.RsWithToolchainTestBase
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.fileTree
+import org.rust.ide.actions.ui.CargoNewCrateSettings
+import org.rust.ide.actions.ui.CargoNewCrateUI
+
+class RsCreateCrateActionTest : RsWithToolchainTestBase() {
+    fun `test create with file context`() {
+        createProjectScaffold()
+
+        val main = cargoProjectDirectory.findFileByRelativePath("Cargo.toml")!!
+        createProject(main, true, "crate1")
```

Comment:
Maybe it's better to make `binary` flag to the end of arguments and pass it in named format?
```suggestion
        createProject(main, "crate1", binary = true)
```
It should improve readability